### ISSUE TITLE
Unify hyperparameters of all frameworks in DCGAN

### ISSUE THREAD

**Issue by astonzhang** [d2l-ai/d2l-en#1778]
https://github.com/d2l-ai/d2l-en/blob/master/chapter_generative-adversarial-networks/dcgan.md

Currently the TF implementation uses a different set of hyperparameters:

#@tab mxnet, pytorch
latent_dim, lr, num_epochs = 100, 0.005, 20
train(net_D, net_G, data_iter, num_epochs, lr, latent_dim)

#@tab tensorflow
latent_dim, lr, num_epochs = 100, 0.0005, 40
train(net_D, net_G, data_iter, num_epochs, lr, latent_dim)

Increasing `num_epochs` to 40 doubles the execution time in TF. Let's unify hyperparameters across all the frameworks.

**Comment by astonzhang** [comment]
@biswajitsahoo1111 Could you take a look at it? Thanks.

**Comment by terrytangyuan** [comment]
Agreed. We should also double check whether the results across all frameworks are consistent when the same set of hyperparameters are used.

**Comment by biswajitsahoo1111** [comment]
@astonzhang Let me try a few other sets of hyperparameters. Getting a common set of hyperparameters should be our goal. I also agree with @terrytangyuan on this.
